src/ui/ui.ts

```diff
@@ -42,14 +42,25 @@ function showPicker() {
 function loadProgram(program: any) {
   programSheet.getRange("A2:G").clearContent();
   const bytecode: string[] = program.data;
-  programSheet
-    .getRange(`A2:B${bytecode.length + 1}`)
-    .setValues(
-      bytecode.map((instruction, i) => [
-        instruction,
-        `=DECODE_INSTRUCTION(A${i + 2})`,
-      ]),
-    );
+  let isConstant: boolean = false;
+  for (var i = 0; i < bytecode.length; i++) {
+    programSheet
+      .getRange(`A${i + 2}:B${i + 2}`)
+      .setValues([
+        [
+          bytecode[i],
+          isConstant
+            ? `=TO_SIGNED_INTEGER(A${i + 2})`
+            : `=DECODE_INSTRUCTION(A${i + 2})`,
+        ],
+      ]);
+    if (!isConstant) {
+      isConstant = size(decodeInstruction(BigInt(bytecode[i]))) == 2;
+    } else {
+      isConstant = false;
+    }
+  }
```

Comment:
when possible, I prefer not to use "stateful" for loops, ie loops where an iteration depends on the previous ones.
```suggestion
  programSheet
    .getRange(`A2:B${bytecode.length + 1}`)
    .setValues(
      bytecode.map((instruction, i) => [
        instruction,
        i > 0 && size(decodeInstruction(BigInt(bytecode[i - 1]))) == 2 ? `=TO_SIGNED_INTEGER(A${i + 2})` : `=DECODE_INSTRUCTION(A${i + 2})`,
      ]),
    );
```

Reply:
of course we compute two times `decodeInstruction` here so it's probably less efficient.